agent/lib/src/adb.dart

```diff
@@ -316,6 +321,34 @@ class IosDeviceDiscovery implements DeviceDiscovery {
   }
 }
 
+class NoOpDeviceDiscovery implements DeviceDiscovery {
```

Comment:
How about naming it as 'vmDeviceDiscovery', and adding comments explaining to bypass device discovery?